wombase_backend/server/apps/tools/views.py

```diff
@@ -103,4 +120,18 @@ def patch(self, request, *args, **kwargs):
 
 class ToolChangesHistoryAPIView(generics.ListAPIView):
     serializer_class = ToolHistorySerializer
-    queryset = Tool.history.filter(history_type="~")
+
+    def get_queryset(self):
+        queryset = Tool.history.filter(history_type="~")
+        for query_param, param_value in self.request.query_params.items():
+            if query_param == "name":
+                queryset = queryset.filter(name__icontains=param_value)
+            if query_param == "category":
+                queryset = queryset.filter(category__name__icontains=param_value)
+            if query_param == "identifier":
+                queryset = queryset.filter(identifier__icontains=param_value)
+            if query_param == "owner":
+                queryset = queryset.filter(owner__id__icontains=param_value)
+            if query_param == "currently_at":
+                queryset = queryset.filter(currently_at__icontains=param_value)
+        return queryset
```

Comment:
for every field except of owner you can just do 

queryset = queryset.filter(**{f"{query_param}__icontains": param_value})

for owner though you need to explicitly state that owner_id__exact=param_value